docs/marxan/data-computations.md

```diff
@@ -0,0 +1,148 @@
+# Cloud platform computations
+
+- Document created on: 27 September 2021.
+- Latest updated on: 27 September 2021.
+
+Throughout the lifecycle of a Marxan Cloud project, the platform carries out
+tabular or geospatial computations at key steps.
+
+This document summarises which kinds of computations are performed, and outlines
+areas for improvement.
+
+## Planning unit grid
+
+When users choose to use a generated planning unit grid (rather than uploading a
+shapefile with a project-specific grid), planning unit geometries are generated
+via PostGIS.
+
+The time complexity of this operation is generally `O(n)`.
+
+## Protected area of planning units
+
+`api/apps/geoprocessing/src/modules/scenario-protected-area-calculation/scenario-protected-area-calculation-processor.ts`
+
+Once users create a scenario, choose relevant IUCN protected area labels or
+upload a shapefile of protected areas, and set the threshold to be used in order
+for a planning unit to be considered as protected, the protected area of each
+planning unit is calculated.
+
+The time complexity of this operation depends on the geometries of the protected
+areas included in the analysis, as well as on the number (and geometries, if
+irregular/user uploaded) of planning units.
+
+## Scenario features
+
+Once users have selected which features they would like to include in the
+analysis, either by picking them as-is or by applying split or stratification
+operations, the geometries of selected features are computed (if stratification
+is applied) or selected (if plainly picked or split, which is equivalent to
+subsetting existing geometries).
+
+When users add new features to the analysis, or configure different split or
+stratification operations for previously selected features, only newly added
+features and features whose geometries have changed (as a consequence of a split
+or stratification operation) are computed anew.
+
+Changes to settings such as `fpf` and `prop` are reflected in the configuration
+of a feature for a scenario, without any geoprocessing operations being
+required.
+
+The level of isolation for computation of features is each individual scenario.
+
+The time complexity of computations of features for a scenario depends largely
+on the geometries of the features as well as on the number (and geometries, if
+irregular/user uploaded) of planning units.
+
+## Marxan input files
```

Comment:
(not related to this particular line) Shouldn't we mention cost-surface upload as well? or it isn't "a computation" ?